### PR TITLE
Changing the timing of key generation from run time to installation time

### PR DESCRIPTION
**Pull request by itouri**
Changing the timing of the key (rsa, ecdsa and ed25519) generation from run time to installation time. The user will decide how to respond when the generated key is lost. For example the user make a back up of the keys.

### CODE REVIEW

**Comment by Metallion** [comment]
* You need to merge origin/master into this branch and resolve the conflict in `sshd.go`.
* What happens when the user deletes the keys in `/etc/openvdc/ssh`?

**Comment by unakatsuo** [comment]
https://ci.openvdc.org/blue/rest/organizations/jenkins/pipelines/citest/branches/executor-hostkey/runs/8/nodes/36/log

```
May 26 11:29:46 executor-lxc vdc-executor[12279]: 2017-05-26 11:29:46 [FATAL] openvdc-executor/main.go:569 Failed to setup SSH Server Error: /etc/openvdc/ssh/host_rsa_key is not a valid pem file: ssh: no key found
                                                  ErrorStackTrace:
                                                  main.(*SSHServer).Setup
                                                          /var/tmp/go/src/github.com/axsh/openvdc/cmd/openvdc-executor/sshd.go:56
                                                  main.execute
                                                          /var/tmp/go/src/github.com/axsh/openvdc/cmd/openvdc-executor/main.go:566
                                                  github.com/axsh/openvdc/vendor/github.com/spf13/cobra.(*Command).execute
                                                          /var/tmp/go/src/github.com/axsh/openvdc/vendor/github.com/spf13/cobra/command.go:636
                                                  github.com/axsh/openvdc/vendor/github.com/spf13/cobra.(*Command).ExecuteC
                                                          /var/tmp/go/src/github.com/axsh/openvdc/vendor/github.com/spf13/cobra/command.go:722
                                                  github.com/axsh/openvdc/vendor/github.com/spf13/cobra.(*Command).Execute
                                                          /var/tmp/go/src/github.com/axsh/openvdc/vendor/github.com/spf13/cobra/command.go:681
                                                  main.main
                                                          /var/tmp/go/src/github.com/axsh/openvdc/cmd/openvdc-executor/main.go:613
                                                  runtime.main
                                                          /usr/lib/golang/src/runtime/proc.go:183
                                                  runtime.goexit
                                                          /usr/lib/golang/src/runtime/asm_amd64.s:2086
```

File: cmd/openvdc-executor/sshd.go [hunk 1]
```diff
@@ -40,31 +35,25 @@ func NewSSHServer(provider hypervisor.HypervisorProvider, ctx context.Context) *
...
-	func(rand io.Reader) (crypto.Signer, error) {
-		return ecdsa.GenerateKey(elliptic.P521(), rand)
-	},
-	func(rand io.Reader) (crypto.Signer, error) {
-		return rsa.GenerateKey(rand, 2048)
-	},
+var KeyGenPathList = []string{
+	"/etc/openvdc/ssh/host_rsa_key",
+	"/etc/openvdc/ssh/host_ecdsa_key",
+	"/etc/openvdc/ssh/host_ed25519_key",
```
Comment (unakatsuo) [inline cmd/openvdc-executor/sshd.go]: I want here to set at the build time. Please refer``build.go`` where overwrite go top level variables.

File: pkg/rhel/openvdc.spec [hunk 2]
```diff
@@ -93,6 +94,12 @@ OpenVDC executor common package.
 /opt/axsh/openvdc/share/mesos-slave/attributes.lxc
 /opt/axsh/openvdc/share/lxc-templates/lxc-openvdc
 %dir /etc/openvdc
+%dir /etc/openvdc/ssh
+
+%post executor
+test ! -f /etc/openvdc/ssh/host_rsa_key && /usr/bin/ssh-keygen -q -t rsa -f /etc/openvdc/ssh/host_rsa_key -C '' -N '' >&/dev/null;
```
Comment (unakatsuo) [inline pkg/rhel/openvdc.spec]: Please confirm the RSA key bit length. Want to set longer than 2048 bit.

**Review by unakatsuo — CHANGES_REQUESTED** [review]

File: pkg/rhel/openvdc.spec (hunk 2 above)
Comment (itouri) [reply pkg/rhel/openvdc.spec]: Default of rsa key size is 2048bit, therefore I'll modify the size of rsa key from 2048 to 4096.

**Review by unakatsuo — APPROVED** [review]

**Pull request merged**